pkg/controller/sparkapplication/controller.go

```diff
@@ -657,6 +657,40 @@ func (c *Controller) submitSparkApplication(app *v1beta2.SparkApplication) *v1be
 		}
 	}
 
+	if c.enableUIService {
+		service, err := createSparkUIService(app, c.kubeClient)
+		if err != nil {
+			glog.Errorf("failed to create UI service for SparkApplication %s/%s: %v", app.Namespace, app.Name, err)
+		} else {
+			app.Status.DriverInfo.WebUIServiceName = service.serviceName
+			app.Status.DriverInfo.WebUIPort = service.servicePort
+			app.Status.DriverInfo.WebUIAddress = fmt.Sprintf("%s:%d", service.serviceIP, app.Status.DriverInfo.WebUIPort)
+			// Create UI Ingress if ingress-format is set.
+			if c.ingressURLFormat != "" {
+				// We are going to want to use an ingress url.
+				ingressURL, err := getSparkUIingressURL(c.ingressURLFormat, app.GetName(), app.GetNamespace())
+				if err != nil {
+					glog.Errorf("failed to run get ingress url %s/%s: %v", app.Namespace, app.Name, err)
+				}
+				// need to ensure the spark.ui variables are configured correctly if a subPath is used.
+				if ingressURL.Path != "" {
+					if app.Spec.SparkConf == nil {
+						app.Spec.SparkConf = make(map[string]string)
+					}
+					app.Spec.SparkConf["spark.ui.proxyBase"] = ingressURL.Path
+					app.Spec.SparkConf["spark.ui.proxyRedirectUri"] = "/"
```

Comment:
might be worth covering this in a test?

Reply:
I've written a test which covers this (sort of)... however, it looks like when I'm writing to SparkConf here, the contents of SparkConf isn't written back to the kubernetes resource. I'm unsure whether it is worth syncing this contents back to the SparkApplication when this has been done